api/src/main/java/org/pmiops/workbench/api/OfflineRdrExportController.java

```diff
@@ -108,18 +109,20 @@ private void groupIdsAndPushTask(List<Long> idList, String taskUri) {
 
   private void createAndPushTask(List<Long> ids, String queuePath, String taskUri) {
     List<String> idsAsString = ids.stream().map(id -> id.toString()).collect(Collectors.toList());
+    Gson gson = new Gson();
+    String daysJson = gson.toJson(ids);
     try (CloudTasksClient client = CloudTasksClient.create()) {
       String commaSepList = String.join(IDS_STRING_SPLIT, idsAsString);
-      Task.Builder taskBuilder =
-          Task.newBuilder()
-              .setAppEngineHttpRequest(
-                  AppEngineHttpRequest.newBuilder()
-                      .setRelativeUri(taskUri)
-                      .setBody(ByteString.copyFrom(commaSepList.getBytes()))
-                      .setHttpMethod(HttpMethod.POST)
-                      .build());
+      AppEngineHttpRequest req =
+          AppEngineHttpRequest.newBuilder()
+              .setRelativeUri(taskUri)
+              .setBody(ByteString.copyFromUtf8(daysJson))
```

Comment:
Is there a reason to do this ByteString conversion? With us moving to JSON, I would expect this to pass in a `json.toString` style thing, rather than a bytestring conversion? Do we need to do this for the AppEngineHttpRequest? (Or will it implicitly handle the conversion from string to encoded string?)

(Also, do you know where the docs for this are? All I've found is https://developers.google.com/resources/api-libraries/documentation/cloudtasks/v2/java/latest/com/google/api/services/cloudtasks/v2/model/AppEngineHttpRequest.html#setBody-java.lang.String- which doesn't have a `putHeaders` method or a builder method)

Reply:
The documentation for appenginetaTask header is pretty bad or not up to date. I reffered to https://cloud.google.com/appengine/docs/standard/python/taskqueue/push/creating-handlers however i could not get a clear way to set header as it is immutable object. After going thru the source code for cloud task i could find this method putHeader

Reply:
Regarding the type, cloud task accept the Body type to be ByteString